web/src/client/software.js

```diff
@@ -52,6 +64,29 @@ const REGISTRATION_IFACE = "org.opensuse.Agama1.Registration";
  * @property {string} message - Result message.
  */
 
+/**
+ * @typedef {object} SoftwareProposal
+ * @property {string} size - Used space in human-readable form.
+ * @property {Object.<string, number>} patterns - Selected patterns and the reason.
+ */
+
+/**
+ * @typedef {object} SoftwareConfig
+ * @property {Array<string>|undefined} patterns - An object where the keys are the pattern names
+ *   and the values whether to install them or not.
+ * @property {string|undefined} product - Product to install.
+ */
+
+/**
+ * @typedef {Object} Pattern
+ * @property {string} name - pattern name (internal ID)
+ * @property {string} category - pattern category
+ * @property {string} summary - pattern name (user visible)
+ * @property {string} description - long description of the pattern
+ * @property {number} order - display order (string!)
+ * @property {string} icon - icon name (not path or file name!)
```

Comment:
NP: different style from the rest, 

```suggestion
 * @property {string} name - Pattern name (internal ID).
 * @property {string} category - Pattern category.
 * @property {string} summary - User visible pattern name.
 * @property {string} description - Long description of the pattern.
 * @property {number} order - Display order (string!).
 * @property {string} icon - Icon name (not path or file name!).
```